hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/resolver/MountTableResolver.java

```diff
@@ -357,7 +357,7 @@ public void refreshEntries(final Collection<MountTable> entries) {
   @VisibleForTesting
   public static boolean isTrashPath(String path) throws IOException {
     Pattern pattern = Pattern.compile(
-        "^" + getTrashRoot() + TRASH_PATTERN + "/");
+        "^" + getTrashRoot() + TRASH_PATTERN);
```

Comment:
Why do we need to remove  `/` ?

Reply:
Fox example , If we don't remove / , client executes command:  ls /user/*/.Trash/Current, it will only return the defalut nameservice  trash path.   
The test case is in org.apache.hadoop.hdfs.server.federation.router.TestRouterTrash#testMultipleMountPoint , 
` FileStatus[] fileStatuses = fs.listStatus(new Path("/user/test-trash/.Trash/Current/"));
    assertEquals(3, fileStatuses.length);`
If we don't remove / , fileStatuses.length will be 1.
Thanks.

Reply:
In  other word, if client listStatus trashpath like '/user/test-trash/.Trash/Current',  this is not consider a trash path
`   if (isTrashPath(path)) {
      path = subtractTrashCurrentPath(path);
    }`

And then it will not find any mountpoint for this trash path,and only return defalut nameservice mountpoint

` public PathLocation getDestinationForPath(final String path)
      throws IOException {
    verifyMountTable();
    PathLocation res;
    readLock.lock();
    try {
      if (this.locationCache == null) {
        res = lookupLocation(processTrashPath(path));
      } else {
        Callable<? extends PathLocation> meh = (Callable<PathLocation>) () -> {
          this.getLocCacheMiss().increment();
          return lookupLocation(processTrashPath(path));
        };
        res = this.locationCache.get(processTrashPath(path), meh);
        this.getLocCacheAccess().increment();
      }
      if (isTrashPath(path)) {
        List<RemoteLocation> remoteLocations = new ArrayList<>();
        for (RemoteLocation remoteLocation : res.getDestinations()) {
          remoteLocations.add(new RemoteLocation(remoteLocation, path));
        }
        return new PathLocation(path, remoteLocations,
            res.getDestinationOrder());
      } else {
        return res;
      }
    } catch (ExecutionException e) {
      Throwable cause = e.getCause();
      final IOException ioe;
      if (cause instanceof IOException) {
        ioe = (IOException) cause;
      } else {
        ioe = new IOException(cause);
      }
      throw ioe;
    } finally {
      readLock.unlock();
    }
  }`